parsimonious/nodes.py

```diff
@@ -99,6 +99,12 @@ def __eq__(self, other):
                 self.end == other.end and
                 self.children == other.children)
 
+    def __hash__(self) -> int:
+        """A unique hash key for this specific Node instance."""
+        return hash(
```

Comment:
I'd like to propose maybe caching the hash value when __hash__ is first called on an instance, this would save 4 the interpreter 4 PyObject_GetAttr calls for each future access, which is not a negligible savings